pkg/kubelet/container/helpers_test.go

```diff
@@ -138,6 +138,84 @@ func TestExpandCommandAndArgs(t *testing.T) {
 	}
 }
 
+func TestExpandVolumeMountsWithSubpath(t *testing.T) {
+	cases := []struct {
+		name            string
+		container       *v1.Container
+		envs            []EnvVar
+		expectedSubPath string
+	}{
+		{
+			name: "none",
+			container: &v1.Container{
+				VolumeMounts: []v1.VolumeMount{{SubPath: "foo"}},
+			},
+			expectedSubPath: "foo",
+		},
+		{
+			name: "volumes expanded",
+			container: &v1.Container{
+				VolumeMounts: []v1.VolumeMount{{SubPath: "foo/$(POD_NAME)"}},
+			},
+			envs: []EnvVar{
+				{
+					Name:  "POD_NAME",
+					Value: "bar",
+				},
+			},
+			expectedSubPath: "foo/bar",
+		},
+	}
+
+	for _, tc := range cases {
+		actualMounts := ExpandContainerVolumeMounts(tc.container, tc.envs)
+
+		if e, a := tc.expectedSubPath, actualMounts[0].SubPath; !reflect.DeepEqual(e, a) {
+			t.Errorf("%v: unexpected subpath; expected %v, got %v", tc.name, e, a)
+		}
+	}
+
+}
+
+func TestExpandVolumeMountsWithoutSubpath(t *testing.T) {
+	cases := []struct {
+		name            string
+		container       *v1.Container
+		envs            []EnvVar
+		expectedSubPath string
+	}{
+		{
+			name: "none",
+			container: &v1.Container{
+				VolumeMounts: []v1.VolumeMount{},
```

Comment:
Sorry, I meant that there's still a VolumeMount object, it just doesn't set subpath.

Reply:
I have pushed a new test with an empty subpath. Hopefully that suits, but if that's still not sufficient, then please let me know.